crates/bevy_feathers/src/controls/button.rs

```diff
@@ -0,0 +1,205 @@
+use bevy_app::{Plugin, PreUpdate};
+use bevy_core_widgets::CoreButton;
+use bevy_ecs::{
+    bundle::Bundle,
+    component::Component,
+    entity::Entity,
+    hierarchy::{ChildOf, Children},
+    lifecycle::RemovedComponents,
+    query::{Added, Changed, Has, Or},
+    schedule::IntoScheduleConfigs,
+    spawn::{SpawnRelated, SpawnableList},
+    system::{Commands, Query, SystemId},
+};
+use bevy_input_focus::tab_navigation::TabIndex;
+use bevy_picking::{hover::Hovered, PickingSystems};
+use bevy_ui::{AlignItems, InteractionDisabled, JustifyContent, Node, Pressed, UiRect, Val};
+use bevy_winit::cursor::CursorIcon;
+
+use crate::{
+    font_styles::InheritableFont,
+    handle_or_path::HandleOrPath,
+    theme::{self, corners::RoundedCorners, fonts, size, ThemeBackgroundColor, ThemeFontColor},
+};
+
+/// Color variants for buttons. This also functions as a component used by the dynamic styling
+/// system to identify which entities are buttons.
+#[derive(Component, Default, Clone)]
+pub enum ButtonVariant {
+    /// The standard button appearance
+    #[default]
+    Normal,
+    /// A button with a more prominent color, this is used for "call to action" buttons,
+    /// default buttons for dialog boxes, and so on.
+    Primary,
+}
+
+/// Parameters for the button template, passed to [`button`] function.
+#[derive(Default, Clone)]
+pub struct ButtonProps {
```

Comment:
I'm surprised that text labels aren't part of these props / spawned with the button. What are your plans there?

Reply:
Labels are just children. The reason for not making the label part of ButtonProps is that a button might not have a label, it might have an icon. Or a label and an icon. Or a drop-down arrow. Or a badge indicating "unread notifications". There are too many possibilities to systematize.